lib/class-wp-rest-block-directory-controller.php

```diff
@@ -262,6 +311,29 @@ public function get_items( $request ) {
 		return rest_ensure_response( $result );
 	}
 
+	/**
+	 * Determine if the endpoints are available.
+	 *
+	 * Only the 'Direct' filesystem transport, and SSH/FTP when credentials are stored are supported at present.
+	 *
+	 * @since 6.5.0
+	 *
+	 * @return bool|WP_Error True if filesystem is available, WP_Error otherwise.
+	 */
+	private static function is_filesystem_available() {
+		$filesystem_method = get_filesystem_method();
+
+		ob_start();
+		$filesystem_credentials_are_stored = request_filesystem_credentials( self_admin_url() );
+		ob_end_clean();
+
+		if ( 'direct' === $filesystem_method || $filesystem_credentials_are_stored ) {
```

Comment:
An easy optimisation would be to put:
```
if ( 'direct' === $filesystem_method ) {
  return true;
}
```
before calling `ob_start()` above.

Reply:
Agreed.